ceph/rbd/workflows/rbd.py

```diff
@@ -264,13 +273,500 @@ def create_pools_and_images(
+            log.error("Multi pool and multi image config failed")
             return 1
 
     # Add back the popped pool test config once configuration is complete
     if pool_test_config:
         multi_pool_config["test_config"] = pool_test_config
 
+    return 0
+
+
+def list_images_and_verify_with_input(**kw):
+    """
+    List all images for in the given pool and verify that the images
+    match the images in the input config
+
+    Args:
+        kw: {
+            "rbd": <>,
+            "pool": <>,
+            "images": <>
+        }
+    """
+    rbd = kw.get("rbd")
+    pool = kw.get("pool")
+    log.info(f"Listing all pools in the cluster of pool type {pool}")
+    out, err = rbd.ls(pool=pool, format="json")
+    if err:
+        log.error(f"Error while fetching images from pool {pool}: {err}")
+        return 1
+    images_in_pool = json.loads(out)
+    images_in_conf = kw.get("images")
+    log.info(f"Images given in input conf: {images_in_conf}")
+    log.info(f"Images listed in pool {pool}: {images_in_pool}")
+    if all(image in images_in_pool for image in images_in_conf):
+        log.info(
+            f"All images given in input conf are also present in images listed in pool {pool}"
+        )
+        return 0
+    else:
+        log.error(f"Images in pool do not match images in input for pool {pool}")
+        return 1
+
+
+def get_image_info_and_verify_for_single_image(**kw):
+    """Get rbd info for the specified image and verify it with input conf
+
+    Args:
+        kw:{
+            "rbd": <>,
+            "pool": <>,
+            "image": <>,
+            "image_conf": <>,
+            "test_ops_parallely": <>
+        }
+    """
+    rbd = kw.get("rbd")
+    pool = kw.get("pool")
+    image = kw.get("image")
+    image_conf = kw.pop("image_conf", {})
+    test_ops_parallely = kw.get("test_ops_parallely", False)
+    log.info(f"Fetching info and verifying for image {image}")
+    out, err = rbd.info(pool=pool, image=image, format="json")
+    if err:
+        log.error(f"Error while fetching image info for image {pool}/{image}: {err}")
+        return 1
+    image_info = json.loads(out)
+    log.info(f"Image info fetched for image {image_info}")
+    log.info(f"Image config passed as input: {image_conf}")
+    req_image_conf = {k: v for k, v in image_conf.items() if k not in ["io_total"]}
+    if not all(keys in image_info.keys() for keys in req_image_conf.keys()):
+        log.error(f"All parameters in {image_conf} are not present in {image_info}")
+        if test_ops_parallely:
+            raise Exception(
+                f"All parameters in {image_conf} are not present in {image_info}"
+            )
+        return 1
+    for k, v in req_image_conf.items():
+        value = image_info[k]
+        if "size" in k:
+            value = convert_size(value)
+        if value != v:
+            log.error(
+                f"Image info for parameter {k} in not same as input config value {v}"
```

Comment:
Depending on whether we are running parallely or serially we should raise exception.
If its sequential, then we can just return value instead of exception.